lib/charms/bind/v0/dns_record.py

```diff
@@ -323,52 +261,73 @@ class RequirerEntry(BaseModel):
     record_data: IPvAnyAddress
     uuid: UUID
 
-    def is_valid(self, dns_domains: List[RequirerDomain]) -> bool:
-        """Validate if the entry has a corresponding domain.
-
-        Args:
-            dns_domains: list of provided DNS domains.
-
-        Returns:
-            true: if there is a matching domain for the entry.
-        """
-        domains = [dns_domain.domain for dns_domain in dns_domains]
-        # pylint doesn't recognise self.domain as a str
-        return any(map(self.domain.endswith, domains))  # pylint: disable=no-member
-
-    def validate_dns_entry(self, info: ValidationInfo) -> "RequirerEntry":
+    def validate_dns_entry(self, _: ValidationInfo) -> "RequirerEntry":
         """Validate DNS entries.
 
-        Args:
-            info: the validation info.
-
         Returns:
             the DNS entry if valid.
-
-        Raises:
-            ValueError: if the DNS entry is not valid.
         """
         validated_entry = RequirerEntry.model_validate(self)
-        if validated_entry.is_valid(info.data["dns_domains"]):
-            return validated_entry
-        raise ValueError(
-            f"Entry with domain {validated_entry.domain} requested without a valid domain"
-        )
+        # Additional validations will be done here
+        return validated_entry
```

Comment:
Just a small reminder that pydantic validation will return not so friendly error to the users unfortunately :/